### PR TITLE
Add darkmode to the standard article layout

### PR DESCRIPTION
**Pull request by abeddow91**
## What does this change?
Adds darkmode colours for a standard article. It also ports across any other use of decide palette by the standard layout to the new themed palette so that, in time, we can remove decide palette. 

> NOTE: Layout stories appear incomplete in dark mode. This is expected. Certain components (eg headline) have not had darkmode implemented yet. Adding these stories now is another way to understand which components still need dark mode. 

## Why?
This is part of a larger body of work to support dark mode on apps.

## Screenshots
Screenshot 2023-11-14 at 13 33 45

### CODE REVIEW

File: dotcom-rendering/scripts/gen-stories/get-stories.mjs [hunk 1]
```diff
@@ -129,11 +129,14 @@ export const ${storyVariableName} = () => {
...
-	design: ArticleDesign.${designName},
-	theme: {...ArticleSpecial, ...Pillar}.${theme.replace('Pillar', '')},
-})];
+${storyVariableName}.decorators = [splitTheme([
+		{
+			display:  ArticleDisplay.${displayName},
+			design: ArticleDesign.${designName},
+			theme: {...ArticleSpecial, ...Pillar}.${theme.replace('Pillar', '')},
+		},
+	]),
```
Comment (mxdvl) [inline dotcom-rendering/scripts/gen-stories/get-stories.mjs:138]: I think that we want these stories to stack? Or maybe even be two separate screenshots?

```suggestion
	], { orientation: 'vertical' }),
```

File: dotcom-rendering/src/lib/decidePalette.ts [hunk 2]
```diff
@@ -517,7 +517,7 @@ const textBranding = (format: ArticleFormat): string => {
 	}
 	return pillarPalette[format.theme].main;
 };
-
+// this has been moved to the theme palette
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/decidePalette.ts:520]:
```suggestion
/** @deprecated this has been moved to the theme palette */
```

**Review by mxdvl — COMMENTED** [review]
This looks terrible! Well done 😆 !

I’m wondering if:
- we should wait until all the elements are converted so we do not accept a “incorrect” chromatic baseline
- we should have some colours handled inside the components, rather that being passed as props
- we should create separate stories for light and dark mode articles, considering how large these screenshots get and the benefits we want to get from diffing.

**Comment by abeddow91** [comment]
Thanks @mxdvl !

- `we should wait until all the elements are converted so we do not accept a “incorrect” chromatic baseline`

Definitely happy to hold off merging this until we have an accurate layout screenshot 👍 having this open as a draft is usefull enough to show what areas / components we've missed / are waiting for.

- `we should have some colours handled inside the components, rather that being passed as props`
I'll take a look if we can refactor some of these. A couple (decideLines and straightLines) are source components so they are probably not good candidates.

- `we should create separate stories for light and dark mode articles, considering how large these screenshots get and the benefits we want to get from diffing.`
Yep, I think this would be valuable. I'll refactor.

File: dotcom-rendering/src/lib/decidePalette.ts [hunk 3]
```diff
@@ -715,6 +716,7 @@ const textCricketScoreboardLink = (): string => {
 	return sport[300];
 };
 
+//This has been moved to the theme palette
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/decidePalette.ts:719]: missed

File: dotcom-rendering/scripts/gen-stories/get-stories.mjs [hunk 4]
```diff
@@ -111,30 +111,60 @@ export const Readme = () => <ReadMe />;
...
+		${storyVariableName + `Dark`}.decorators = [darkDecorator(
+				{
+					display:  ArticleDisplay.${displayName},
+					design: ArticleDesign.${designName},
+					theme: {...ArticleSpecial, ...Pillar}.${theme.replace('Pillar', '')},
+				}
+			),
+		];`
+		}
+	`;
```
Comment (mxdvl) [inline dotcom-rendering/scripts/gen-stories/get-stories.mjs:167]: I don’t think you want this to have indentation

File: dotcom-rendering/scripts/gen-stories/get-stories.mjs [hunk 5]
```diff
@@ -55,7 +55,7 @@ const LAYOUT_TEMPLATE_HEADER = `
  * These stories were auto-generated by \`dotcom-rendering/scripts/gen-stories/gen-stories.js\`
  */
 import { ArticleDesign, ArticleDisplay, ArticleSpecial, Pillar } from '@guardian/libs';
-import { lightDecorator } from '../../.storybook/decorators/themeDecorator';
+import { lightDecorator, darkDecorator } from '../../.storybook/decorators/themeDecorator';
```
Comment (mxdvl) [inline dotcom-rendering/scripts/gen-stories/get-stories.mjs:58]: If you want your import order to be correct…

```suggestion
import { darkDecorator, lightDecorator } from '../../.storybook/decorators/themeDecorator';
```

**Review by mxdvl — APPROVED** [review]

File: dotcom-rendering/src/palette.ts [hunk 6]
```diff
@@ -316,25 +315,134 @@ const starRatingBackgroundColourLight = (): string =>
 const starRatingBackgroundColourDark = (): string =>
 	sourcePalette.brandAlt[200];
 
-const blockQuoteFillLight = (format: ArticleFormat): string => {
-	return decidePalette(format).fill.blockquoteIcon;
+const blockQuoteFillLight = ({ design, theme }: ArticleFormat): string => {
```
Comment (ioannakok) [inline dotcom-rendering/src/palette.ts:318]: Think these are already in main from https://github.com/guardian/dotcom-rendering/pull/9469. Not sure why you're getting this diff

**Review by ioannakok — APPROVED** [review]
Looks great!

**Pull request merged**